### PR TITLE
Onboarding Microsoft.Authorization (authorization specs only) for autogeneration

### PR DESCRIPTION
**Pull request by dagoroz**
```
> azure-schema-generator@1.0.0 generate-single C:\Repos\azure-schemas\generator
> ts-node cmd/generatesingle "authorization/resource-manager"

[C:\Users\daorozco\AppData\Local\Temp\schm_azspc] executing: git fsck
[C:\Users\daorozco\AppData\Local\Temp\schm_azspc] executing: git reset -q .
[C:\Users\daorozco\AppData\Local\Temp\schm_azspc] executing: git checkout -q -- .
[C:\Users\daorozco\AppData\Local\Temp\schm_azspc] executing: git clean -q -fd
[C:\Users\daorozco\AppData\Local\Temp\schm_azspc] executing: git gc
[C:\Users\daorozco\AppData\Local\Temp\schm_azspc] executing: git fetch -q
[C:\Users\daorozco\AppData\Local\Temp\schm_azspc] executing: git checkout -q origin/master
Using autogenlist config:
{
  "basePath": "authorization/resource-manager",
  "namespace": "Microsoft.Authorization",
  "resourceConfig": [
    {
      "type": "roleAssignments",
      "scopes": 62
    },
    {
      "type": "roleDefinitions",
      "scopes": 62
    },
    {
      "type": "roleAssignmentScheduleRequests",
      "scopes": 62
    },
    {
      "type": "roleEligibilityScheduleRequests",
      "scopes": 62
    },
    {
      "type": "roleManagementPolicyAssignments",
      "scopes": 62
    },
    {
      "type": "roleAssignmentApprovals/stages",
      "scopes": 2
    }
  ],
  "suffix": "Authorization"
}
[C:\Repos\azure-schemas\generator] executing: autorest.cmd --version=3.0.6374 --use=@autorest/azureresourceschema@3.0.92 --azureresourceschema --output-folder=C:\Users\daorozco\AppData\Local\Temp\z00xb8n6yr --multiapi --pass-thru:subset-reducer C:\Users\daorozco\AppData\Local\Temp\schm_azspc\specification\authorization\resource-manager\readme.md
AutoRest code generation utility [cli version: 3.1.4; node: v14.15.4, max-memory: 4096 MB]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest

There is a new version of AutoRest available (3.2.3).
 > You can install the newer version with with npm install -g autorest@latest

   Loading AutoRest core      'C:\Users\daorozco\.autorest\@autorest_core@3.0.6374\node_modules\@autorest\core\dist' (3.0.6374)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.92->3.0.92)
Processing batch task - {"tag":"schema-2021-03-01-preview"} .
Processing batch task - {"tag":"schema-2020-10-01-preview"} .
WARNING: Skipping path '/{roleAssignmentId}': Unable to locate '/providers/' segment
Processing batch task - {"tag":"schema-2015-06-01"} .
Processing batch task - {"tag":"schema-2015-07-01"} .
WARNING: Skipping path '/{roleAssignmentId}': Unable to locate '/providers/' segment
Processing batch task - {"tag":"schema-2018-01-01-preview"} .
WARNING: Skipping path '/{roleId}': Unable to locate '/providers/' segment
Processing batch task - {"tag":"schema-2018-07-01-preview"} .
Processing batch task - {"tag":"schema-2019-08-01-preview"} .
Processing batch task - {"tag":"schema-2020-08-01-preview"} .
WARNING: Skipping path '/{roleAssignmentId}': Unable to locate '/providers/' segment
Processing batch task - {"tag":"schema-2020-04-01-preview"} .
WARNING: Skipping path '/{roleId}': Unable to locate '/providers/' segment
Processing batch task - {"tag":"schema-2020-03-01-preview"} .
WARNING: Skipping path '/{roleAssignmentId}': Unable to locate '/providers/' segment
Processing batch task - {"tag":"schema-2018-09-01-preview"} .
WARNING: Skipping path '/{roleId}': Unable to locate '/providers/' segment
Processing batch task - {"tag":"schema-2018-05-01-preview"} .
Processing batch task - {"tag":"schema-2017-10-01-preview"} .
WARNING: Skipping path '/{roleAssignmentId}': Unable to locate '/providers/' segment
Processing batch task - {"tag":"schema-2021-01-01-preview"} .
[28.44 s] Generation Complete
================================================================================================================================
Filename: C:\Repos\azure-schemas\schemas\2015-07-01\Microsoft.Authorization.Authorization.json
Provider Namespace: Microsoft.Authorization
API Version: 2015-07-01
Resource Types (Tenant Scope):
- roleAssignments
- roleDefinitions
Resource Types (Management Group Scope):
- roleAssignments
- roleDefinitions
Resource Types (Subscription Scope):
- roleAssignments
- roleDefinitions
Resource Types (Resource Group Scope):
- roleAssignments
- roleDefinitions
Resource Types (Extension Scope):
- roleAssignments
- roleDefinitions
================================================================================================================================
================================================================================================================================
Filename: C:\Repos\azure-schemas\schemas\2017-10-01-preview\Microsoft.Authorization.Authorization.json
Provider Namespace: Microsoft.Authorization
API Version: 2017-10-01-preview
Resource Types (Tenant Scope):
- roleAssignments
Resource Types (Management Group Scope):
- roleAssignments
Resource Types (Subscription Scope):
- roleAssignments
Resource Types (Resource Group Scope):
- roleAssignments
Resource Types (Extension Scope):
- roleAssignments
================================================================================================================================
================================================================================================================================
Filename: C:\Repos\azure-schemas\schemas\2018-01-01-preview\Microsoft.Authorization.Authorization.json
Provider Namespace: Microsoft.Authorization
API Version: 2018-01-01-preview
Resource Types (Tenant Scope):
- roleAssignments
- roleDefinitions
Resource Types (Management Group Scope):
- roleAssignments
- roleDefinitions
Resource Types (Subscription Scope):
- roleAssignments
- roleDefinitions
Resource Types (Resource Group Scope):
- roleAssignments
- roleDefinitions
Resource Types (Extension Scope):
- roleAssignments
- roleDefinitions
================================================================================================================================
================================================================================================================================
Filename: C:\Repos\azure-schemas\schemas\2018-05-01-preview\Microsoft.Authorization.Authorization.json
Provider Namespace: Microsoft.Authorization
API Version: 2018-05-01-preview
Resource Types (Subscription Scope):
- accessReviewScheduleDefinitions
- accessReviewScheduleSettings
================================================================================================================================
================================================================================================================================
Filename: C:\Repos\azure-schemas\schemas\2018-09-01-preview\Microsoft.Authorization.Authorization.json
Provider Namespace: Microsoft.Authorization
API Version: 2018-09-01-preview
Resource Types (Tenant Scope):
- roleAssignments
Resource Types (Management Group Scope):
- roleAssignments
Resource Types (Subscription Scope):
- roleAssignments
Resource Types (Resource Group Scope):
- roleAssignments
Resource Types (Extension Scope):
- roleAssignments
================================================================================================================================
================================================================================================================================
Filename: C:\Repos\azure-schemas\schemas\2020-03-01-preview\Microsoft.Authorization.Authorization.json
Provider Namespace: Microsoft.Authorization
API Version: 2020-03-01-preview
Resource Types (Tenant Scope):
- roleAssignments
Resource Types (Management Group Scope):
- roleAssignments
Resource Types (Subscription Scope):
- roleAssignments
Resource Types (Resource Group Scope):
- roleAssignments
Resource Types (Extension Scope):
- roleAssignments
================================================================================================================================
================================================================================================================================
Filename: C:\Repos\azure-schemas\schemas\2020-04-01-preview\Microsoft.Authorization.Authorization.json
Provider Namespace: Microsoft.Authorization
API Version: 2020-04-01-preview
Resource Types (Tenant Scope):
- roleAssignments
Resource Types (Management Group Scope):
- roleAssignments
Resource Types (Subscription Scope):
- roleAssignments
Resource Types (Resource Group Scope):
- roleAssignments
Resource Types (Extension Scope):
- roleAssignments
================================================================================================================================
================================================================================================================================
Filename: C:\Repos\azure-schemas\schemas\2020-08-01-preview\Microsoft.Authorization.Authorization.json
Provider Namespace: Microsoft.Authorization
API Version: 2020-08-01-preview
Resource Types (Tenant Scope):
- roleAssignments
Resource Types (Management Group Scope):
- roleAssignments
Resource Types (Subscription Scope):
- roleAssignments
Resource Types (Resource Group Scope):
- roleAssignments
Resource Types (Extension Scope):
- roleAssignments
================================================================================================================================
================================================================================================================================
Filename: C:\Repos\azure-schemas\schemas\2020-10-01-preview\Microsoft.Authorization.Authorization.json
Provider Namespace: Microsoft.Authorization
API Version: 2020-10-01-preview
Resource Types (Tenant Scope):
- roleAssignments
- roleAssignmentScheduleRequests
- roleEligibilityScheduleRequests
- roleManagementPolicyAssignments
Resource Types (Management Group Scope):
- roleAssignments
- roleAssignmentScheduleRequests
- roleEligibilityScheduleRequests
- roleManagementPolicyAssignments
Resource Types (Subscription Scope):
- roleAssignments
- roleAssignmentScheduleRequests
- roleEligibilityScheduleRequests
- roleManagementPolicyAssignments
Resource Types (Resource Group Scope):
- roleAssignments
- roleAssignmentScheduleRequests
- roleEligibilityScheduleRequests
- roleManagementPolicyAssignments
Resource Types (Extension Scope):
- roleAssignments
- roleAssignmentScheduleRequests
- roleEligibilityScheduleRequests
- roleManagementPolicyAssignments
================================================================================================================================
================================================================================================================================
Filename: C:\Repos\azure-schemas\schemas\2021-01-01-preview\Microsoft.Authorization.Authorization.json
Provider Namespace: Microsoft.Authorization
API Version: 2021-01-01-preview
Resource Types (Tenant Scope):
- roleAssignmentApprovals/stages
================================================================================================================================
================================================================================================================================
Filename: C:\Repos\azure-schemas\schemas\2021-03-01-preview\Microsoft.Authorization.Authorization.json
Provider Namespace: Microsoft.Authorization
API Version: 2021-03-01-preview
Resource Types (Subscription Scope):
- accessReviewScheduleDefinitions
- accessReviewScheduleSettings
================================================================================================================================

```

### CODE REVIEW

**Comment by dagoroz** [comment]
@jorgecotillo @anthony-c-martin can I have a review, please? :)

**Comment by dagoroz** [comment]
@jorgecotillo @anthony-c-martin ping

File: generator/autogenlist.ts [hunk 1]
```diff
@@ -56,6 +56,37 @@ const autoGenList: AutoGenConfig[] = [
...
+    {
+        basePath: 'authorization/resource-manager',
+        namespace: 'Microsoft.Authorization',
+        resourceConfig: [
+            {
+                type: 'roleAssignments',
+                scopes:ScopeType.Extension
+            },
+            {
+                type:'roleDefinitions',
```
Comment (anthony-c-martin) [inline generator/autogenlist.ts:68]: Same question for "roleDefinitions"

File: generator/autogenlist.ts [hunk 2]
```diff
@@ -56,6 +56,37 @@ const autoGenList: AutoGenConfig[] = [
...
+            {
+                type: 'roleEligibilityScheduleRequests',
+                scopes:ScopeType.Extension
+            },
+            {
+                type: 'roleManagementPolicyAssignments',
+                scopes:ScopeType.Extension
+            },
+            {
+                type: 'roleAssignmentApprovals/stages',
```
Comment (anthony-c-martin) [inline generator/autogenlist.ts:84]: This looks like it's being used at the tenant scope in the swagger examples: https://github.com/Azure/azure-rest-api-specs/blob/1b0ed8edd58bb7c9ade9a27430759527bd4eec8e/specification/authorization/resource-manager/Microsoft.Authorization/preview/2021-01-01-preview/examples/GetRoleAssignmentApprovals.json#L10

**Review by anthony-c-martin — CHANGES_REQUESTED** [review]
See questions about the scope configurations

File: generator/autogenlist.ts [hunk 3]
```diff
@@ -56,6 +56,37 @@ const autoGenList: AutoGenConfig[] = [
         basePath: 'automation/resource-manager',
         namespace: 'Microsoft.Automation',
     },
+    {
+        basePath: 'authorization/resource-manager',
+        namespace: 'Microsoft.Authorization',
+        resourceConfig: [
+            {
+                type: 'roleAssignments',
+                scopes:ScopeType.Extension
```
[the remark this answers is not in the document]
Comment (dagoroz) [reply generator/autogenlist.ts:65]: Corrected in next iteration

**Comment by dagoroz** [comment]
@anthony-c-martin are we good? or do you need me to do something else from my side?
I'm not sure how onboarding happens from your perspective hence the question 😊

**Comment by dagoroz** [comment]
@jorgecotillo @anthony-c-martin ping

**Pull request closed without merging**